src/Navigation.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+
+import { NavigationProps } from '../typings/react-day-picker';
+import { getNavigation } from './helpers/getNavigation';
+import { getNavigationProps } from './helpers/getNavigationProps';
+
+export const Navigation: React.FC<NavigationProps> = props => {
+  const { dayPickerProps } = props;
+  const {
+    onMonthChange,
+    onPrevClick,
+    onNextClick,
+    onStartClick,
+    startDay,
+  } = dayPickerProps;
+
+  const { nextMonth, prevMonth } = getNavigation(dayPickerProps);
+
+  const {
+    containerProps,
+    nextProps,
+    prevProps,
+    startProps,
+  } = getNavigationProps(dayPickerProps);
+
+  const handlePrevClick = (e: React.MouseEvent<HTMLButtonElement>): void => {
+    onMonthChange && prevMonth && onMonthChange(prevMonth, e);
+    onPrevClick && prevMonth && onPrevClick(prevMonth, e);
+  };
+  const handleNextClick = (e: React.MouseEvent<HTMLButtonElement>): void => {
+    onMonthChange && nextMonth && onMonthChange(nextMonth, e);
+    onNextClick && nextMonth && onNextClick(nextMonth, e);
+  };
+  const handleStartClick = (e: React.MouseEvent<HTMLButtonElement>): void => {
```

Comment:
Does assigning function like this create new references ever render? I've always defined functions like so...
```
export const MyComp: React.FC<MyCompProps> = () => {
	// ... implementation
	
	return <div onClick={handleOnClick} />;

	function handleOnClick() {
		// handler
	}
};
```

Reply:
Probably goes along with your performance optimization response from my other comment... carr yon

Reply:
That is an interesting solution, do you have more insights about that pattern?

Anyway, yes it creates new references. We may want to use `useCallback` if worth it (but I don't think it is).

Reply:
I just read through [this article](https://kentcdodds.com/blog/usememo-and-usecallback) that has a really good explanation of what's going on here.

In summary, `useMemo` is going to be better, but the optimizations aren't worth the code complexity in this scenario. 

As far as defining a function scoped to the component (like my example) and what you have, I haven't really found any articles or done extensive perf testing myself. I think it stems from the `eslint(react/jsx-no-bind)` rule I usually code with that yells at me.